src/main/java/dev/ftb/mods/ftbessentials/command/TeleportCommands.java

```diff
@@ -125,11 +135,24 @@ private static TeleportPos findBlockPos(ServerLevel world, ServerPlayer player,
 		}
 
 		world.getChunk(currentPos.getX() >> 4, currentPos.getZ() >> 4, ChunkStatus.HEIGHTMAPS);
-		BlockPos newPos = world.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, currentPos);
-
-		if (newPos.getY() > 0) {
-			player.displayClientMessage(new TextComponent(String.format("Found good location after %d " + (attempt == 1 ? "attempt" : "attempts") + " @ [x %d, z %d]", attempt, newPos.getX(), newPos.getZ())), false);
-			return new TeleportPos(world.dimension(), newPos.above());
+		BlockPos hmPos = world.getHeightmapPos(Heightmap.Types.MOTION_BLOCKING_NO_LEAVES, currentPos);
+
+		if (hmPos.getY() > 0) {
+			if (hmPos.getY() >= world.getHeight()) { // broken heightmap (nether, other mod dimensions)
+				Iterator<BlockPos.MutableBlockPos> it = BlockPos.spiralAround(new BlockPos(hmPos.getX(), world.getSeaLevel(), hmPos.getY()), 16, Direction.EAST, Direction.SOUTH).iterator();
+				while (it.hasNext()) {
+					BlockPos newPos = it.next();
+					BlockState bs = world.getBlockState(newPos);
+
+					if (bs.getMaterial().isSolidBlocking() && !bs.is(BlockTags.LEAVES) && !bs.is(Blocks.BEDROCK) && isBlockPosAir(world, newPos.above(1)) && isBlockPosAir(world, newPos.above(2)) && isBlockPosAir(world, newPos.above(3))) {
```

Comment:
Hm, this should use `ftbessentials:ignore_rtp` block tag that includes these two

Reply:
I can't find ignore_rtp tag, where is it defined?

Reply:
You should define the tag yourself (using `TagHooks.getBlockOptional`) and just add the leaves tag and bedrock as values to it by default ^^ This enables server admins to configure custom blocks they don't want players to be rtp'd on top of

Reply:
I'm really not the person to throw tasks at. It wasn't there previously aswell. Maintainer push is enabled on this PR.